In [1]:
import numpy as np
import pandas as pd

1. **LOWERCASE**

In [2]:
df=pd.read_csv("/content/IMDB Dataset.csv")

In [3]:
df.head()

review sentiment
0  One of the other reviewers has mentioned that ...  positive
1  A wonderful little production. <br /><br />The...  positive
2  I thought this was a wonderful way to spend ti...  positive
3  Basically there's a family where a little boy ...  negative
4  Petter Mattei's "Love in the Time of Money" is...  positive

In [4]:
df.shape

(51648, 2)

In [5]:
df['review'][3]

"Basically there's a family where a little boy (Jake) thinks there's a zombie in his closet & his parents are fighting all the time.<br /><br />This movie is slower than a soap opera... and suddenly, Jake decides to become Rambo and kill the zombie.<br /><br />OK, first of all when you're going to make a film you must Decide if its a thriller or a drama! As a drama the movie is watchable. Parents are divorcing & arguing like in real life. And then we have Jake with his closet which totally ruins all the film! I expected to see a BOOGEYMAN similar movie, and instead i watched a drama with some meaningless thriller spots.<br /><br />3 out of 10 just for the well playing parents & descent dialogs. As for the shots with Jake: just ignore them."

In [6]:
df['review'][3].lower #lowercases 3rd review

<function str.lower()>

In [7]:
df['review'].str.lower() #lowercases all reviews

0        one of the other reviewers has mentioned that ...
1        a wonderful little production. <br /><br />the...
2        i thought this was a wonderful way to spend ti...
3        basically there's a family where a little boy ...
4        petter mattei's "love in the time of money" is...
                               ...                        
51643    i thought this movie did a down right good job...
51644    bad plot, bad dialogue, bad acting, idiotic di...
51645    i am a catholic taught in parochial elementary...
51646    i'm going to have to disagree with the previou...
51647    no one expects the star trek movies to be high...
Name: review, Length: 51648, dtype: object

**2. Removing HTML TAGS**

In [8]:
import re
def remove_html_tags(text):
  pattern=re.compile('<.*?>')
  return pattern.sub(r'',text)

In [9]:
text="<p>This is <br>a paragraph<br> with line breaks.</p>"

In [10]:
remove_html_tags(text)

'This is a paragraph with line breaks.'

In [11]:
df['review'].apply(remove_html_tags)

0        One of the other reviewers has mentioned that ...
1        A wonderful little production. The filming tec...
2        I thought this was a wonderful way to spend ti...
3        Basically there's a family where a little boy ...
4        Petter Mattei's "Love in the Time of Money" is...
                               ...                        
51643    I thought this movie did a down right good job...
51644    Bad plot, bad dialogue, bad acting, idiotic di...
51645    I am a Catholic taught in parochial elementary...
51646    I'm going to have to disagree with the previou...
51647    No one expects the Star Trek movies to be high...
Name: review, Length: 51648, dtype: object

 **3. Removing URLs**

In [12]:
import re
def remove_url(text):
  pattern=re.compile(r'https?://\S+|www\.\S+')
  return pattern.sub(r'',text)

In [13]:
text1='Check https://leandomainsearch.com/'
text2='Click on this http //innovative india.mygov.in/ppc2023'
text3='Google link www.google.com'

In [14]:
remove_url(text1)

'Check '

 **4. Removing Punctuations**

In [15]:
import string, time
string.punctuation #gives punctuation set

'!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'

In [16]:
exclude=string.punctuation

In [17]:
import re
def remove_punc1(text):
    return text.translate(str.maketrans('', '', exclude))
 

In [18]:
text='In each town—London, Paris and Rome—we stayed in youth hostels.'

In [19]:
remove_punc1(text)

'In each town—London Paris and Rome—we stayed in youth hostels'

In [20]:
df1=pd.read_csv("/content/twitter sentiment analysis.csv")

In [21]:
df1.sample(5)

id  label                                              tweet
17722  17723      0  fingers crossed my new shoes arrive tomorrow.....
13497  13498      0  @user #us has enough to worry about! #includin...
18967  18968      0    does cheryl need a relationship to be happy?...
18784  18785      0  i'm like to sleep over the jok #fun #funk #fun...
15472  15473      0  bubble run 🌏🐳💜💙💛💗#2 . . . #b...

In [22]:
df1.shape

(31962, 3)

In [23]:
df1['tweet']=df1['tweet'].apply(remove_punc1)

In [24]:
df1

id  label                                              tweet
0          1      0   user when a father is dysfunctional and is so...
1          2      0  user user thanks for lyft credit i cant use ca...
2          3      0                                bihday your majesty
3          4      0  model   i love u take with u all the time in u...
4          5      0               factsguide society now    motivation
...      ...    ...                                                ...
31957  31958      0  ate user isz that youuu😍😍😍😍😍ðŸ˜...
31958  31959      0    to see nina turner on the airwaves trying to...
31959  31960      0  listening to sad songs on a monday morning otw...
31960  31961      1  user sikh temple vandalised in in calgary wso ...
31961  31962      0                    thank you user for you follow  

[31962 rows x 3 columns]

**5**. **Removing** **Punctuations**

In [25]:
from textblob import TextBlob #textblob handles common type spelling mistakes

In [26]:
incorrect_text='the hotel wass too smal ro acomodate the larg partty'
textBlb=TextBlob(incorrect_text)
textBlb.correct().string

'the hotel was too small to accommodate the large party'

5. Removing Stop words

In [27]:
import nltk
nltk.download('stopwords')
from nltk.corpus import stopwords
sw_nltk=stopwords.words('english')

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [28]:
sw_nltk

['i',
 'me',
 'my',
 'myself',
 'we',
 'our',
 'ours',
 'ourselves',
 'you',
 "you're",
 "you've",
 "you'll",
 "you'd",
 'your',
 'yours',
 'yourself',
 'yourselves',
 'he',
 'him',
 'his',
 'himself',
 'she',
 "she's",
 'her',
 'hers',
 'herself',
 'it',
 "it's",
 'its',
 'itself',
 'they',
 'them',
 'their',
 'theirs',
 'themselves',
 'what',
 'which',
 'who',
 'whom',
 'this',
 'that',
 "that'll",
 'these',
 'those',
 'am',
 'is',
 'are',
 'was',
 'were',
 'be',
 'been',
 'being',
 'have',
 'has',
 'had',
 'having',
 'do',
 'does',
 'did',
 'doing',
 'a',
 'an',
 'the',
 'and',
 'but',
 'if',
 'or',
 'because',
 'as',
 'until',
 'while',
 'of',
 'at',
 'by',
 'for',
 'with',
 'about',
 'against',
 'between',
 'into',
 'through',
 'during',
 'before',
 'after',
 'above',
 'below',
 'to',
 'from',
 'up',
 'down',
 'in',
 'out',
 'on',
 'off',
 'over',
 'under',
 'again',
 'further',
 'then',
 'once',
 'here',
 'there',
 'when',
 'where',
 'why',
 'how',
 'all',
 'any',
 'both',
 'each

In [29]:
print(len(sw_nltk))#179 stopwords in nltk

179


In [30]:
def remove_stopwords(text):
    new_text = []
    
    for word in text.split():
        if word in stopwords.words('english'):
            new_text.append('')
        else:
            new_text.append(word)
    x = new_text[:]
    new_text.clear()
    return " ".join(x)

In [31]:
remove_stopwords('probably my all-time favorite movie, a story of selflessness, sacrifice and dedication to a noble cause, but it\'s not preachy or boring. it just never gets old, despite my having seen it some 15 or more times')

'probably  all-time favorite movie,  story  selflessness, sacrifice  dedication   noble cause,    preachy  boring.   never gets old, despite   seen   15   times'

In [32]:
df.head()

review sentiment
0  One of the other reviewers has mentioned that ...  positive
1  A wonderful little production. <br /><br />The...  positive
2  I thought this was a wonderful way to spend ti...  positive
3  Basically there's a family where a little boy ...  negative
4  Petter Mattei's "Love in the Time of Money" is...  positive

In [33]:
df1['tweet'].apply(remove_stopwords)

0        user   father  dysfunctional    selfish  drags...
1        user user thanks  lyft credit  cant use cause ...
2                                          bihday  majesty
3        model  love u take  u   time  ur📱 😙😎ð...
4                           factsguide society  motivation
                               ...                        
31957    ate user isz  youuu😍😍😍😍😍😍ðŸ˜...
31958     see nina turner   airwaves trying  wrap    ma...
31959    listening  sad songs   monday morning otw  wor...
31960    user sikh temple vandalised   calgary wso cond...
31961                                 thank  user   follow
Name: tweet, Length: 31962, dtype: object

In [34]:
!python3 -m spacy download en_core_web_sm

/usr/local/lib/python3.8/dist-packages/torch/cuda/__init__.py:497: UserWarning: Can't initialize NVML
  warnings.warn("Can't initialize NVML")
2023-01-18 02:33:27.330452: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.8/12.8 MB 62.4 MB/s eta 0:00:00
✔ Download and installation successful
You can now load the package via spacy.load('en_core_web_sm')


In [35]:
import spacy
#loading english language small model of spacy
en=spacy.load('en_core_web_sm')
sw_spacy=en.Defaults.stop_words
print(sw_spacy)

/usr/local/lib/python3.8/dist-packages/torch/cuda/__init__.py:497: UserWarning: Can't initialize NVML
  warnings.warn("Can't initialize NVML")


{'seems', 'myself', 'please', 'even', 'between', 'though', 'fifty', 'also', 'someone', 'hence', 'did', 'next', 'of', 'often', 'formerly', 'such', 'ca', 'ten', 'other', 'besides', 'can', 'own', 'unless', 'wherein', 'by', 'something', 'under', 'do', 'already', 'anyone', 'give', '‘s', '’s', 'almost', 'take', 'over', 'front', 'hereafter', 'some', 'might', 'very', 'whereafter', 'that', 'move', 'this', 'to', 'it', 'more', 'again', 'about', 'therein', 'whereby', 'via', 'with', 'up', 'whatever', 'ever', 'twenty', "n't", 'them', 'he', 'name', 'fifteen', 'being', 'whereas', 'whether', 'before', 'moreover', 'nobody', 'various', 'latter', 'most', 'seeming', 'toward', 'should', 'which', 'could', 'these', 'although', 'through', 'whoever', 'but', 'show', 'therefore', 'everywhere', 'quite', 'too', 'at', 'out', 'due', 'really', 'who', 'never', 'above', 'themselves', 'ours', 'always', '‘m', 'until', 'where', 'hereby', 'two', 'few', 'full', 'thereupon', 'less', 'six', 'itself', 'side', 'becoming', 'is', 

In [36]:
print(len(sw_spacy))

326


In [37]:
import gensim
from gensim.parsing.preprocessing import remove_stopwords, STOPWORDS
print(STOPWORDS)

frozenset({'seems', 'myself', 'please', 'even', 'between', 'didn', 'ie', 'though', 'also', 'someone', 'hence', 'did', 'next', 'hasnt', 'of', 'often', 'formerly', 'such', 'ten', 'etc', 'other', 'besides', 'own', 'can', 'wherein', 'unless', 'by', 'something', 'under', 'do', 'already', 'anyone', 'give', 'almost', 'take', 'over', 'front', 'hereafter', 'some', 'might', 'very', 'whereafter', 'that', 'move', 'un', 'to', 'this', 'it', 'more', 'again', 'eg', 'about', 'therein', 'whereby', 'via', 'with', 'up', 'whatever', 'ever', 'twenty', 'them', 'name', 'he', 'fifteen', 'being', 'whereas', 'whether', 'before', 'moreover', 'nobody', 'various', 'latter', 'most', 'de', 'seeming', 'km', 'toward', 'should', 'thin', 'could', 'which', 'these', 'although', 'don', 'through', 'whoever', 'but', 'show', 'therefore', 'everywhere', 'cant', 'quite', 'detail', 'too', 'at', 'kg', 'out', 'due', 'really', 'who', 'never', 'above', 'themselves', 'doesn', 'ours', 'always', 'until', 'where', 'hereby', 'two', 'few', 

In [38]:
print(len(STOPWORDS))

337


179 stopwords in nltk-mostly used

326 stopwords in spacy

337 STOPWORDS in gensism-rarely used

In [39]:
sw_nltk.extend(['first','second'])#Extending stopwords list
print(len(sw_nltk))

181


In [40]:
sw_nltk.remove('now') #Removing stopwords from set
print(len(sw_nltk))

180


6. Handling emojis: Removing/Replacing

In [41]:
import re
def remove_emoji(text):
    emoji_pattern = re.compile("["
                           u"\U0001F600-\U0001F64F"  # emoticons
                           u"\U0001F300-\U0001F5FF"  # symbols & pictographs
                           u"\U0001F680-\U0001F6FF"  # transport & map symbols
                           u"\U0001F1E0-\U0001F1FF"  # flags (iOS)
                           u"\U00002702-\U000027B0"
                           u"\U000024C2-\U0001F251"
                           "]+", flags=re.UNICODE)
    return emoji_pattern.sub(r'', text)

In [42]:
remove_emoji("Loved the movie. It was 😘😘")

'Loved the movie. It was '

In [43]:
pip install emoji 

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [44]:
import emoji
print(emoji.demojize('Python is 🔥'))

Python is :fire:


In [45]:
print(emoji.demojize('Loved the movie. It was 😘'))

Loved the movie. It was :face_blowing_a_kiss:


#Tokenization Techniques

1. Using the split function

In [46]:
# word tokenization
sent1 = 'I am going to delhi'
sent1.split()

['I', 'am', 'going', 'to', 'delhi']

In [47]:
# sentence tokenization
sent2 = 'I am going to delhi. I will stay there for 3 days. Let\'s hope the trip to be great'
sent2.split('.')

['I am going to delhi',
 ' I will stay there for 3 days',
 " Let's hope the trip to be great"]

In [48]:
# Problems with split function
sent3 = 'I am going to delhi!'
sent3.split()

['I', 'am', 'going', 'to', 'delhi!']

In [49]:
sent4 = 'Where do think I should go? I have 3 day holiday'
sent4.split('.')

['Where do think I should go? I have 3 day holiday']

2. Regular Expression

In [50]:
import re
sent3 = 'I am going to delhi!'
tokens = re.findall("[\w']+", sent3)
tokens

['I', 'am', 'going', 'to', 'delhi']

In [51]:
text = """Lorem Ipsum is simply dummy text of the printing and typesetting industry? 
Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, 
when an unknown printer took a galley of type and scrambled it to make a type specimen book."""
sentences = re.compile('[.!?] ').split(text)
sentences

['Lorem Ipsum is simply dummy text of the printing and typesetting industry',
 "\nLorem Ipsum has been the industry's standard dummy text ever since the 1500s, \nwhen an unknown printer took a galley of type and scrambled it to make a type specimen book."]

3. NLTK & Keras-Tokenizaton-Word & Sentence Tokenize

In [52]:
from nltk.tokenize import word_tokenize,sent_tokenize
import nltk
nltk.download('punkt')

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.


True

In [53]:
sent1 = 'I am going to visit delhi!'
word_tokenize(sent1)

['I', 'am', 'going', 'to', 'visit', 'delhi', '!']

In [54]:
sent5 = 'I have a Ph.D in A.I'
sent6 = "We're here to help! mail us at nks@gmail.com"
sent7 = 'A 5km ride cost $10.50'
paragraph = 'Bali is predominantly a Hindu country. Bali is known for its elaborate, traditional dancing. The dancing is inspired by its Hindi beliefs. Most of the dancing portrays tales of good versus evil. To watch the dancing is a breathtaking experience. Lombok has some impressive points of interest – the majestic Gunung Rinjani is an active volcano. It is the second highest peak in Indonesia. Art is a Balinese passion. Batik paintings and carved statues make popular souvenirs. Artists can be seen whittling and painting on the streets, particularly in Ubud. It is easy to appreciate each island as an attractive tourist destination. Majestic scenery; rich culture; white sands and warm, azure waters draw visitors like magnets every year. Snorkelling and diving around the nearby Gili Islands is magnificent. Marine fish, starfish, turtles and coral reef are present in abundance. Bali and Lombok are part of the Indonesian archipelago. Bali has some spectacular temples. The most significant is the Mother Temple, Besakih. The inhabitants of Lombok are mostly Muslim with a Hindu minority. Lombok remains the most understated of the two islands. Lombok has several temples worthy of a visit, though they are less prolific. Bali and Lombok are neighbouring islands.'
word_tokenize(sent5)

['I', 'have', 'a', 'Ph.D', 'in', 'A.I']

In [55]:
word_tokenize(sent6)

['We',
 "'re",
 'here',
 'to',
 'help',
 '!',
 'mail',
 'us',
 'at',
 'nks',
 '@',
 'gmail.com']

In [56]:
word_tokenize(sent7)

['A', '5km', 'ride', 'cost', '$', '10.50']

In [57]:
sent_tokenize(paragraph)

['Bali is predominantly a Hindu country.',
 'Bali is known for its elaborate, traditional dancing.',
 'The dancing is inspired by its Hindi beliefs.',
 'Most of the dancing portrays tales of good versus evil.',
 'To watch the dancing is a breathtaking experience.',
 'Lombok has some impressive points of interest – the majestic Gunung Rinjani is an active volcano.',
 'It is the second highest peak in Indonesia.',
 'Art is a Balinese passion.',
 'Batik paintings and carved statues make popular souvenirs.',
 'Artists can be seen whittling and painting on the streets, particularly in Ubud.',
 'It is easy to appreciate each island as an attractive tourist destination.',
 'Majestic scenery; rich culture; white sands and warm, azure waters draw visitors like magnets every year.',
 'Snorkelling and diving around the nearby Gili Islands is magnificent.',
 'Marine fish, starfish, turtles and coral reef are present in abundance.',
 'Bali and Lombok are part of the Indonesian archipelago.',
 'Bali

In [58]:
from keras.preprocessing.text import Tokenizer
from keras.preprocessing.text import text_to_word_sequence

In [59]:
tokenizer=Tokenizer()
tokenizer.fit_on_texts(paragraph)
train_sequence=text_to_word_sequence(paragraph)
print(train_sequence)

['bali', 'is', 'predominantly', 'a', 'hindu', 'country', 'bali', 'is', 'known', 'for', 'its', 'elaborate', 'traditional', 'dancing', 'the', 'dancing', 'is', 'inspired', 'by', 'its', 'hindi', 'beliefs', 'most', 'of', 'the', 'dancing', 'portrays', 'tales', 'of', 'good', 'versus', 'evil', 'to', 'watch', 'the', 'dancing', 'is', 'a', 'breathtaking', 'experience', 'lombok', 'has', 'some', 'impressive', 'points', 'of', 'interest', '–', 'the', 'majestic', 'gunung', 'rinjani', 'is', 'an', 'active', 'volcano', 'it', 'is', 'the', 'second', 'highest', 'peak', 'in', 'indonesia', 'art', 'is', 'a', 'balinese', 'passion', 'batik', 'paintings', 'and', 'carved', 'statues', 'make', 'popular', 'souvenirs', 'artists', 'can', 'be', 'seen', 'whittling', 'and', 'painting', 'on', 'the', 'streets', 'particularly', 'in', 'ubud', 'it', 'is', 'easy', 'to', 'appreciate', 'each', 'island', 'as', 'an', 'attractive', 'tourist', 'destination', 'majestic', 'scenery', 'rich', 'culture', 'white', 'sands', 'and', 'warm', '

4. Spacy

In [60]:
import spacy
nlp = spacy.load('en_core_web_sm')

In [61]:
doc1 = nlp(sent5) #Convering sentence to document
doc2 = nlp(sent6)
doc3 = nlp(sent7)
doc4 = nlp(sent1)

In [62]:
for token in doc4:
    print(token)

I
am
going
to
visit
delhi
!


#Stemming

In [63]:
from nltk.stem.porter import PorterStemmer

In [64]:
ps = PorterStemmer()
def stem_words(text):
    return " ".join([ps.stem(word) for word in text.split()])

In [65]:
sample = "walk walks walking walked"
stem_words(sample)

'walk walk walk walk'

In [66]:
text = 'probably my alltime favorite movie a story of selflessness sacrifice and dedication to a noble cause but its not preachy or boring it just never gets old despite my having seen it some 15 or more times in the last 25 years paul lukas performance brings tears to my eyes and bette davis in one of her very few truly sympathetic roles is a delight the kids are as grandma says more like dressedup midgets than children but that only makes them more fun to watch and the mothers slow awakening to whats happening in the world and under her own roof is believable and startling if i had a dozen thumbs theyd all be up for this movie'
print(text)

probably my alltime favorite movie a story of selflessness sacrifice and dedication to a noble cause but its not preachy or boring it just never gets old despite my having seen it some 15 or more times in the last 25 years paul lukas performance brings tears to my eyes and bette davis in one of her very few truly sympathetic roles is a delight the kids are as grandma says more like dressedup midgets than children but that only makes them more fun to watch and the mothers slow awakening to whats happening in the world and under her own roof is believable and startling if i had a dozen thumbs theyd all be up for this movie


In [67]:
stem_words(text)

'probabl my alltim favorit movi a stori of selfless sacrific and dedic to a nobl caus but it not preachi or bore it just never get old despit my have seen it some 15 or more time in the last 25 year paul luka perform bring tear to my eye and bett davi in one of her veri few truli sympathet role is a delight the kid are as grandma say more like dressedup midget than children but that onli make them more fun to watch and the mother slow awaken to what happen in the world and under her own roof is believ and startl if i had a dozen thumb theyd all be up for thi movi'

Another way

In [68]:
sentences=nltk.sent_tokenize(paragraph)
stemmer=PorterStemmer()

In [69]:
for i in range(len(sentences)):
  words = nltk.word_tokenize(sentences[i])#converts into words
  words=[stemmer.stem(word) for word in words if word not in set(stopwords.words('english'))]
  #For each & every word is not in the set of stopwords 
  sentences[i]= ''.join(words)#after getting words, joining the words

In [70]:
sentences

['balipredominantlihinducountri.',
 'baliknownelabor,traditdanc.',
 'thedancinspirhindibelief.',
 'mostdancportraytalegoodversuevil.',
 'towatchdancbreathtakexperi.',
 'lombokimpresspointinterest–majestgunungrinjaniactivvolcano.',
 'itsecondhighestpeakindonesia.',
 'artbalinespassion.',
 'batikpaintcarvstatumakepopularsouvenir.',
 'artistseenwhittlpaintstreet,particularliubud.',
 'iteasiappreciislandattracttouristdestin.',
 'majestsceneri;richcultur;whitesandwarm,azurwaterdrawvisitorlikemagneteveriyear.',
 'snorkeldivearoundnearbigiliislandmagnific.',
 'marinfish,starfish,turtlcoralreefpresentabund.',
 'balilombokpartindonesianarchipelago.',
 'balispectaculartempl.',
 'thesignificmothertempl,besakih.',
 'theinhabitlombokmostlimuslimhinduminor.',
 'lombokremainundersttwoisland.',
 'lomboksevertemplworthivisit,thoughlessprolif.',
 'balilombokneighbourisland.']

#Lemmatization

In [71]:
from nltk.stem import WordNetLemmatizer
nltk.download('wordnet')
nltk.download('omw-1.4')
sentences=nltk.sent_tokenize(paragraph)
lemmatizer= WordNetLemmatizer()

[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data] Downloading package omw-1.4 to /root/nltk_data...


In [72]:
for i in range(len(sentences)):
  words = nltk.word_tokenize(sentences[i])#converts into words
  words=[lemmatizer.lemmatize(word) for word in words if word not in set(stopwords.words('english'))]
  sentences[i]= ''.join(words)

In [73]:
 sentences

['BalipredominantlyHinducountry.',
 'Baliknownelaborate,traditionaldancing.',
 'ThedancinginspiredHindibelief.',
 'Mostdancingportraystalegoodversusevil.',
 'Towatchdancingbreathtakingexperience.',
 'Lombokimpressivepointinterest–majesticGunungRinjaniactivevolcano.',
 'ItsecondhighestpeakIndonesia.',
 'ArtBalinesepassion.',
 'Batikpaintingcarvedstatuemakepopularsouvenir.',
 'Artistsseenwhittlingpaintingstreet,particularlyUbud.',
 'Iteasyappreciateislandattractivetouristdestination.',
 'Majesticscenery;richculture;whitesandwarm,azurewaterdrawvisitorlikemagneteveryyear.',
 'SnorkellingdivingaroundnearbyGiliIslandsmagnificent.',
 'Marinefish,starfish,turtlecoralreefpresentabundance.',
 'BaliLombokpartIndonesianarchipelago.',
 'Balispectaculartemple.',
 'ThesignificantMotherTemple,Besakih.',
 'TheinhabitantLombokmostlyMuslimHinduminority.',
 'Lombokremainsunderstatedtwoisland.',
 'Lombokseveraltempleworthyvisit,thoughleprolific.',
 'BaliLombokneighbouringisland.']

In [74]:
import nltk
from nltk.stem import WordNetLemmatizer
import nltk
nltk.download('wordnet')
nltk.download('omw-1.4')
wordnet_lemmatizer = WordNetLemmatizer()

sentence = "He was running and eating at same time. He has bad habit of swimming after playing long hours in the Sun."
punctuations="?:!.,;"
sentence_words = nltk.word_tokenize(sentence)
for word in sentence_words:
    if word in punctuations:
        sentence_words.remove(word)

sentence_words
print("{0:20}{1:20}".format("Word","Lemma"))
for word in sentence_words:
    print ("{0:20}{1:20}".format(word,wordnet_lemmatizer.lemmatize(word,pos='v')))#v=verb

Word                Lemma               
He                  He                  
was                 be                  
running             run                 
and                 and                 
eating              eat                 
at                  at                  
same                same                
time                time                
He                  He                  
has                 have                
bad                 bad                 
habit               habit               
of                  of                  
swimming            swim                
after               after               
playing             play                
long                long                
hours               hours               
in                  in                  
the                 the                 
Sun                 Sun                 


[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package omw-1.4 to /root/nltk_data...
[nltk_data]   Package omw-1.4 is already up-to-date!
